[Chore](job)Provides configuration of job execution queue size

CalvinKirs opened the pull request:
## Proposed changes

When dealing with a large number of tasks, the default execution queue size is 1024. This can lead to tasks being dropped if the queue becomes full. 
eg

`dispatch instant task failed, job id is xxx`


To address this, you can add the parameters `insert_task_queue_size` and `mtmv_task_queue_size` in the `fe.conf` configuration file. These parameters must be set to a power of 2. 

**Keep in mind, increasing this value is recommended only when thread resources are limited; otherwise, you should consider increasing the number of task execution threads.**

CalvinKirs committed:
[Chore](job)Provides configuration of job execution queue size

When dealing with a large number of tasks, the default execution queue size is 1024. This can lead to tasks being dropped if the queue becomes full. To address this, you can add the parameters `insert_task_queue_size` and `mtmv_task_queue_size` in the `fe.conf` configuration file. These parameters must be set to a power of 2. Keep in mind, increasing this value is recommended only when thread resources are limited; otherwise, you should consider increasing the number of task execution threads.

CalvinKirs commented:
run buildall

CalvinKirs committed:
[Chore](job)Provides configuration of job execution queue size

When dealing with a large number of tasks, the default execution queue size is 1024. This can lead to tasks being dropped if the queue becomes full. To address this, you can add the parameters `insert_task_queue_size` and `mtmv_task_queue_size` in the `fe.conf` configuration file. These parameters must be set to a power of 2. Keep in mind, increasing this value is recommended only when thread resources are limited; otherwise, you should consider increasing the number of task execution threads.

morningman commented on fe/fe-common/src/main/java/org/apache/doris/common/Config.java:
```diff
@@ -1776,6 +1776,12 @@ public class Config extends ConfigBase {
             "Maximum number of persistence allowed per task in a job,exceeding which old tasks will be discarded，"
                    + "If the value is less than 1, it will not be persisted." })
     public static int max_persistence_task_count = 100;
+    @ConfField(description = {"MV task 的等待队列大小，值必须是 2 的幂，否则将使用默认值 1024", "The size of the MV task's"
```
use `setter` to check if value is power of 2.
And you can also check if the value is too large or too small.
And need to test unit test

morningman commented on fe/fe-core/src/main/java/org/apache/doris/job/manager/TaskDisruptorGroupManager.java:
```diff
@@ -65,8 +65,10 @@ public class TaskDisruptorGroupManager<T extends AbstractTask> {
 
-    private static final int DISPATCH_INSERT_TASK_QUEUE_SIZE = DEFAULT_RING_BUFFER_SIZE;
-    private static final int DISPATCH_MTMV_TASK_QUEUE_SIZE = DEFAULT_RING_BUFFER_SIZE;
+    private static final int DISPATCH_INSERT_TASK_QUEUE_SIZE = isPowerOfTwo(Config.insert_task_queue_size)
```
Do not check `isPowerOfTwo` here

CalvinKirs committed:
[Chore](job)Provides configuration of job execution queue size

When dealing with a large number of tasks, the default execution queue size is 1024. This can lead to tasks being dropped if the queue becomes full. To address this, you can add the parameters `insert_task_queue_size` and `mtmv_task_queue_size` in the `fe.conf` configuration file. These parameters must be set to a power of 2. Keep in mind, increasing this value is recommended only when thread resources are limited; otherwise, you should consider increasing the number of task execution threads.

CalvinKirs commented:
run buildall

CalvinKirs committed:
[Chore](job)Provides configuration of job execution queue size

When dealing with a large number of tasks, the default execution queue size is 1024. This can lead to tasks being dropped if the queue becomes full. To address this, you can add the parameters `insert_task_queue_size` and `mtmv_task_queue_size` in the `fe.conf` configuration file. These parameters must be set to a power of 2. Keep in mind, increasing this value is recommended only when thread resources are limited; otherwise, you should consider increasing the number of task execution threads.

morningman reviewed (approved):
LGTM

CalvinKirs commented:
run buildall

CalvinKirs commented:
run cloudp0

CalvinKirs commented:
run cloud_p0
